File: other-programming-languages/Ocaml/2024/2024-001/part-001/main.ml
(* Read the input file and parse it into two lists *)
let read_file file_path =
  let left_list = ref [] in
  let right_list = ref [] in
  let ic = open_in file_path in
  try
    while true do
      let line = input_line ic in
      (* Trim leading and trailing whitespace *)
      let line = String.trim line in
      if line <> "" then
        match String.split_on_char ' ' line |> List.filter (fun s -> s <> "") with
        | [left; right] ->
            left_list := int_of_string left :: !left_list;
            right_list := int_of_string right :: !right_list
        | _ -> failwith "Malformed line in input file"
    done;
    (!left_list, !right_list)
  with End_of_file ->
    close_in ic;
    (!left_list, !right_list)

(* Function to calculate total distance *)
let calculate_total_distance left_list right_list =
  let sorted_left = List.sort compare left_list in
  let sorted_right = List.sort compare right_list in
  List.fold_left2 (fun acc left right -> acc + abs (left - right)) 0 sorted_left sorted_right

(* Main function *)
let () =
  let file_path = "input.txt" in
  let left_list, right_list = read_file file_path in
  let total_distance = calculate_total_distance left_list right_list in
  Printf.printf "Total Distance: %d\n" total_distance

File: other-programming-languages/PHP/2024/2024-001/part-001/main.php
<?php
// Define the file path
$filePath = 'input.txt';

// Check if the file exists
if (!file_exists($filePath)) {
    die("File not found: " . $filePath);
}

// Load the file content into an array
$fileContent = file($filePath, FILE_IGNORE_NEW_LINES | FILE_SKIP_EMPTY_LINES);

// Initialize the left and right lists
$leftList = [];
$rightList = [];

// Parse the file content
foreach ($fileContent as $line) {
    $columns = preg_split('/\s+/', trim($line));
    if (count($columns) >= 2) {
        $leftList[] = (int)$columns[0];
        $rightList[] = (int)$columns[1];
    }
}

// Sort both lists
sort($leftList);
sort($rightList);

// Calculate the total distance
$totalDistance = 0;
for ($i = 0; $i < min(count($leftList), count($rightList)); $i++) {
    $totalDistance += abs($leftList[$i] - $rightList[$i]);
}

// Output the result
echo "Total Distance: " . $totalDistance;
?>
